File: models/account.go
package models

type Account struct {
	Name        string `json:"name" binding:"required"`
	DisplayName string `json:"displayName" binding:"required"`
	Uuid        int    `json:"uuid" binding:"required"`
	IconPath    string `json:"iconPath" binding:"required"`
	Password    string `json:"password" binding:"required"`
}

File: www/assets/js/model/Account.ts
export default class Account {
    public name: string | null = null
    public displayName: string | null = null
    public password: string | null = null
    public currentPassword: string | null = null

    public toJson(): string {
        const obj = {
            name: this.name,
            displayName: this.displayName,
            password: this.password,
            currentPassword: this.currentPassword,
        }

        return JSON.stringify(obj)
    }
}